src/Storages/StorageFile.cpp

```diff
@@ -791,6 +804,30 @@ StorageFile::StorageFile(const std::string & table_path_, const std::string & us
     setStorageMetadata(args);
 }
 
+StorageFile::StorageFile(
```

Comment:
Here and in other places: we should look at the case with a single path as a case with Array of the length 1 to avoid code duplication and to automatically test the current implementation on all tests that we have.